src/SpaceVecAlg/AdmittanceVec.h

```diff
@@ -0,0 +1,181 @@
+
+#pragma once
+
+namespace sva
+{
+
+using namespace Eigen;
+
+/**
+	* Admittance Vector.
+	* Mechanical admittance is a measure of how much a structure moves
+	* in response to a given force:
+	*
+	*     v = A * F
+	*
+	* where F is the exerted force, A the admittance and v the resulting output
+	* velocity. Admittance is the reciprocal of mechanical impedance; see
+	* <https://en.wikipedia.org/wiki/Mechanical_impedance>.
+	*/
+template<typename T>
+class AdmittanceVec
+{
+public:
+	typedef Vector3<T> vector3_t;
+	typedef Vector6<T> vector6_t;
+
+	friend class PTransform<T>;
+
+public:
+	/// Zero admittance vector
+	static AdmittanceVec<T> Zero()
+	{
+		return AdmittanceVec<T>(vector3_t::Zero(), vector3_t::Zero());
+	}
+
+public:
+	AdmittanceVec():
+		angular_(),
+		linear_()
+	{}
+
+	/**
+		* @param vec Admittance vector with angular motion in head
+		* and linear motion in tail.
+		*/
+	AdmittanceVec(const vector6_t& vec):
+		angular_(vec.template head<3>()),
+		linear_(vec.template tail<3>())
+	{}
+
+	/**
+		* @param angular Angular admittance.
+		* @param linear Linear admittance.
+		*/
+	AdmittanceVec(const vector3_t& angular, const vector3_t& linear):
+		angular_(angular),
+		linear_(linear)
+	{}
+
+	/** Homogeneous admittance constructor.
+		* @param angular Angular admittance.
+		* @param linear Linear admittance.
+		*/
+	AdmittanceVec(T angular, T linear):
+		angular_(angular, angular, angular),
+		linear_(linear, linear, linear)
+	{}
+
+	// Accessor
+	/// @return Angular admittance
+	const vector3_t& angular() const
+	{
+		return angular_;
+	}
+
+	/// @return Angular admittance
+	vector3_t& angular()
+	{
+		return angular_;
+	}
+
+	/// @return Linear admittance
+	const vector3_t& linear() const
+	{
+		return linear_;
+	}
+
+	/// @return Linear admittance
+	vector3_t& linear()
+	{
+		return linear_;
+	}
+
+	/// @return Non-compact admittance vector.
+	vector6_t vector() const
+	{
+		return ((vector6_t() << angular_, linear_).finished());
+	}
+
+	template<typename T2>
+	AdmittanceVec<T2> cast() const
+	{
+		return AdmittanceVec<T2>(angular_.template cast<T2>(), linear_.template cast<T2>());
+	}
+
+	// Operators
+	AdmittanceVec<T> operator+(const AdmittanceVec<T>& av) const
+	{
+		return AdmittanceVec<T>(angular_ + av.angular_, linear_ + av.linear_);
+	}
+
+	AdmittanceVec<T>& operator+=(const AdmittanceVec<T>& av)
+	{
+		angular_ += av.angular_;
+		linear_ += av.linear_;
+		return *this;
+	}
+
+	template<typename T2>
+	AdmittanceVec<T> operator*(T2 scalar) const
+	{
+		return AdmittanceVec<T>(scalar*angular_, scalar*linear_);
+	}
+
+	template<typename T2>
+	AdmittanceVec<T> operator/(T2 scalar) const
+	{
+		return AdmittanceVec<T>(angular_/scalar, linear_/scalar);
+	}
+
+	bool operator==(const AdmittanceVec<T>& av) const
+	{
+		return (angular_ == av.angular_) && (linear_ == av.linear_);
+	}
+
+	bool operator!=(const AdmittanceVec<T>& av) const
+	{
+		return (angular_ != av.angular_) || (linear_ != av.linear_);
```

Comment:
`return !(*this == av);`

Reply:
Done in c1aea17de83c14922ec1499d46666ef017a44a76